datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -487,7 +487,8 @@ public Map<String, List<DatastreamTask>> getTasksToCleanUp(List<DatastreamGroup>
     Map<String, DatastreamTask> assignmentsMap = currentAssignment.values()
         .stream()
         .flatMap(Collection::stream)
-        .collect(Collectors.toMap(DatastreamTask::getDatastreamTaskName, Function.identity()));
+        .collect(Collectors.toMap(DatastreamTask::getDatastreamTaskName, Function.identity(),
+            (existingTask, duplicateTask) -> existingTask));
```

Comment:
Those tasks will be duplicates. The previous leader when rearranging tasks, would initially copy a particular task to another host and then remove that task from the assigned host. The clean up is meant to be done in case the leader couldn't finish that for every rearrangements, and that is when this situation would occur. 

Hence the partition assignments within those tasks would be the same as well.